numpy/core/src/umath/loops.c.src

```diff
@@ -840,6 +847,62 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 void
     UNARY_LOOP_FAST(@type@, @type@, *out = in > 0 ? 1 : (in < 0 ? -1 : 0));
 }
 
+#ifndef USE_LEGACY_DIVISION
+NPY_NO_EXPORT void
+@TYPE@_divide(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
+{
+    static const struct libdivide_@div@_t EmptyStruct;
+    BINARY_LOOP_BASE
+
+    if(!is2) {
+        const @type@ in2 = *(@type@ *)ip2;
+        struct libdivide_@div@_t fast_d = in2 ? libdivide_@div@_gen(in2) : EmptyStruct;
+        BINARY_LOOP_FIXED {
+            const @type@ in1 = *(@type@ *)ip1;
+            /*
+             * FIXME: On x86 at least, dividing the smallest representable integer
+             * by -1 causes a SIFGPE (division overflow). We treat this case here
+             * (to avoid a SIGFPE crash at python level), but a good solution would
+             * be to treat integer division problems separately from FPU exceptions
+             * (i.e. a different approach than npy_set_floatstatus_divbyzero()).
+             */
+            if (in2 == 0 || (in1 == NPY_MIN_@TYPE@ && in2 == -1)) {
+                npy_set_floatstatus_divbyzero();
+                *((@type@ *)op1) = 0;
+            }
+            else if (((in1 > 0) != (in2 > 0)) && (in1 % in2 != 0)) {
```

Comment:
Can `libdivide` compute `in1 % in2` for you? It seems a bit silly to use libdivide only to then perform a remainder calculation without it.

Reply:
I honestly think we can avoid this `%` and rewrite it as postprocessing?

Reply:
So I still don't know how removing the `%` gave no performance boot :). The compiler is magically optimizing something. https://github.com/numpy/numpy/pull/17727#issuecomment-723538185

Reply:
Oh, interesting. @ganesh-k13 two things: First make sure you are dividing a positive by a negative number (or vice versa), otherwise this is not hit at all. Second, was the timing difference with libdivide? I guess it might be the compiler is smart enough to optimize the modulo away, but I would be surprised if it is smart enough when libdivide is being used?

Reply:
They seem to have not done it yet: https://github.com/ridiculousfish/libdivide/issues/9

Reply:
All this changes is subtract one for rounding purproses, Now unless there is some edge case again, I think you can just do without the subtract, and then move the if to later, so that `if (res < 0) && (res * in2 != in1) { res -= 1}`?

Reply:
You were right about not hitting the case, @seberg , seems like in the profile script I forgot to invert the signs. Above method seems to work, few edge cases to iron out(like <= 0, etc), will try them.

Reply:
I found three edge cases:
1. res is `0`, then possible negative divisor/dividend
2. divisor is 0 or -0, handled by putting inside else
3. dividend is 0, handled by the same logic as 1.

Let me know if any more are there.